### PR TITLE
DOCS-1376: Document `close()` method for all components

### PR DESCRIPTION
**Pull request by sguequierre**
* Adds both Python and Go

All services to come

### CODE REVIEW

File: docs/components/input-controller/_index.md [hunk 1]
```diff
@@ -447,7 +447,9 @@ The [motion](/services/motion/) and [navigation](/services/navigation/) services
 For more information, see the [Python SDK Docs](https://python.viam.dev/autoapi/viam/components/input/client/index.html#viam.components.input.client.ControllerClient.get_geometries).
 
 ```python {class="line-numbers linkable-line-numbers"}
-my_controller = Controller.from_robot(robot=robot, name="my_controller")
+my_controller = Controller.from_robot(
+    robot=myRobotWithController,
```
Comment (sguequierre) [inline docs/components/input-controller/_index.md:451]:
```suggestion
    robot=myRobotWithController,
```

File: docs/components/encoder/_index.md [hunk 2]
```diff
@@ -20,7 +20,7 @@ The encoder component supports:
 
 - [Incremental encoders](https://en.wikipedia.org/wiki/Incremental_encoder#Quadrature_outputs), which can measure the speed and direction of rotation in relation to a given reference point like a starting point.
   These encoders output two phases.
-  Based on the sequence and timing of these phases, it is determined how far something has turned and in which direction.
+  Encoderd on the sequence and timing of these phases, it is determined how far something has turned and in which direction.
```
Comment (npentrel) [inline docs/components/encoder/_index.md:23]:
```suggestion
  Based on the sequence and timing of these phases, it is determined how far something has turned and in which direction.
```

**Review by npentrel — APPROVED** [review]
LGTM % one suggestion

File: docs/components/encoder/_index.md (hunk 2 above)
Comment (sguequierre) [reply docs/components/encoder/_index.md:23]: oops-- TY

**Pull request merged**